doc/source/how-to-use-driver-api.rst

```diff
@@ -0,0 +1,254 @@
+
+.. code-block:: python
+
+    # File: server.py
+
+    def pytorch_to_parameter_record(pytorch_module: torch.nn.Module):
+        state_dict = pytorch_module.state_dict()
+        for k, v in state_dict.items():
+            state_dict[k] = _ndarray_to_array(v.numpy())
+        return ParametersRecord(state_dict)
+
+
+    @app.main()
+    def main(driver: Driver, context: Context) -> None:
+    	global_model = Net()
+
+Then, we create a :code:`RecordSet` and add parameters and configurations to :code:`parameters_record` and :code:`configs_record`, respectively. Note that the dictionary keys used here are customizable, so you have a great degree of flexibility to assign and use the dictionaries in a :code:`RecordSet`:
+
+.. code-block:: python
+
+    # File: server.py
+    # In the for-loop in `main()`
+
+    # Create a RecordSet
+    recordset = RecordSet()
+
+    # Add model parameters to the RecordSet
+    recordset.parameters_records["my_model"] = pytorch_to_parameter_record(global_model)
+
+    # Add a training configuration for 1 epoch only to the RecordSet
+    recordset.configs_records["my_config"] = ConfigsRecord({"epochs": 1})
+
+Next, we create a list of :code:`Messages`, one for each node ID. To do so, we loop over all node IDs and run :code:`Driver.create_message()` with the :code:`recordset` as the content of the message: 
+
+.. code-block:: python
+
+    # File: server.py
+
+    messages = []
+    for node_id in node_ids:
+        message = driver.create_message(
+            content=recordset,
+            message_type=MessageType.TRAIN,
+            dst_node_id=node_id,
+            group_id=str(server_round),
+            ttl=DEFAULT_TTL,
+        )
+        messages.append(message)
+
+Finally, we use :code:`Driver.push_messages()` to push the list of :code:`Messages` containing the encapsulated parameters and configurations to the nodes.
+
+.. code-block:: python
+
+    # File: server.py
+
+    message_ids = driver.push_messages(messages)
+
+:code:`Driver.push_messages()` yields an iterable list of message IDs. In some real-world scenarios, you may encounter situations where only some :code:`Messages` can be pushed, so it is good practice to filter out empty message IDs:
+
+.. code-block:: python
+
+    # File: server.py
+
+    # Wait for results, ignore empty message_ids
+    message_ids = [message_id for message_id in message_ids if message_id != ""]
+
+
+Pull messages from nodes
+------------------------
+
+Once messages are successfully sent to the nodes, we can use the associated message IDs to get results from these nodes. To do so, we continuously run :code:`Driver.pull_messages()` with the list of :code:`message_ids` until all of the :code:`Messages` from the nodes are received. 
```

Comment:
I think it's worth mentioning that what's happening here is that the `ServerApp` is requesting the `SuperLink` for any result from the `Messages` previously pushed (of which we have retained the IDs).